### PR TITLE
store selected event ids in the url and load the events on refresh

### PR DESCRIPTION
**Pull request by AndreMarkert**

### CODE REVIEW

File: app.vue [hunk 1]
```diff
@@ -275,4 +275,32 @@ const downloadCalendar = () => {
...
+  const selectedEventIds = [...selectedEvents.values()].map(
+    (event) => event.id
+  );
+  const url = new URL(window.location.href);
+  if (selectedEventIds.length === 0) {
+    url.searchParams.delete("events");
+  } else {
+    url.searchParams.set("events", selectedEventIds.join(","));
+  }
+  window.history.replaceState({}, "", url.toString());
```
Comment (SebastianSiedler) [inline app.vue]: why don't you use `vue-router` instead of manipulating the history stack on your own?

File: fiwClient.ts [hunk 2]
```diff
@@ -107,6 +107,25 @@ export const client = {
     return SingleEventSchema.parse(response.data);
   },
 
+  // fetch multiple events by their ids
+  getEventsByIds: async (ids: number[]) => {
+    // load all events in parallel
+    const responses = await Promise.all(
+      ids.map(async (id) => {
+        try {
+          const response = await axiosInstance.get(`/events/${id}`);
```
Comment (SebastianSiedler) [inline fiwClient.ts]: dublicate code.  
why don't you just use the already existing getEventById()?

File: fiwClient.ts [hunk 3]
```diff
@@ -107,6 +107,25 @@ export const client = {
     return SingleEventSchema.parse(response.data);
   },
 
+  // fetch multiple events by their ids
+  getEventsByIds: async (ids: number[]) => {
```
Comment (SebastianSiedler) [inline fiwClient.ts]: this code should be located in the component using this method. Not in the fiw http client.

File: app.vue [hunk 4]
```diff
@@ -275,4 +275,32 @@ const downloadCalendar = () => {
...
+});
+
+// load all valid events that are in the URL
+onMounted(() => {
```
Comment (SebastianSiedler) [inline app.vue]: use `vue-router` and zod to get and parse the parameters

Comment (SebastianSiedler) [inline app.vue]: also i think you don't need the onMounted. Just running this code in the `setup` should be fine

File: app.vue [hunk 5]
```diff
@@ -275,4 +275,32 @@ const downloadCalendar = () => {
...
   return new Date(date).toLocaleString("de");
 };
+
+// add selected events to URL
+watch(selectedEvents, () => {
+  const selectedEventIds = [...selectedEvents.values()].map(
+    (event) => event.id
+  );
+  const url = new URL(window.location.href);
+  if (selectedEventIds.length === 0) {
```
Comment (SebastianSiedler) [inline app.vue]: use vue-router for manipulating the url

**Review by SebastianSiedler — CHANGES_REQUESTED** [review]

File: fiwClient.ts (hunk 2 above)
Comment (AndreMarkert) [reply fiwClient.ts]: Fixed

File: fiwClient.ts (hunk 3 above)
Comment (AndreMarkert) [reply fiwClient.ts]: Fixed

File: app.vue (hunk 4 above)
Comment (AndreMarkert) [reply app.vue]: Fixed

File: app.vue (hunk 5 above)
Comment (AndreMarkert) [reply app.vue]: Fixed

File: app.vue (hunk 4 above)
Comment (AndreMarkert) [reply app.vue]: Fixed

File: app.vue (hunk 1 above)
Comment (AndreMarkert) [reply app.vue]: Fixed

**Pull request merged**